### DISCUSSION TITLE
Problems with External Data Connector in Automation

### DISCUSSION THREAD

**Discussion by rhrish** [Budibase/budibase#2950] [Help]
First of all, thanks for the update! The 'State' feature is super useful for the current application I am building!

I am using Mariadb for my application and I have a couple of stored procedures that I call from the application. For each of these procedures, I have a query(from the 'Add Query' option on the 'Data' page) created in my app, along with a couple of parameters that I can pass from the front end of the app.

After the recent update to 0.9.161, I have been trying to add an 'External Data Connector' to my automation step. 
This is what happens. 
- I click on Setup
- The Query dropdown shows all the queries I've created
- I try and select a query, the choice goes back to 'Choose an option'

The second problem I've been trying to figure is the 'Finish and Test automation' button.
- When I click on this button, it shows a modal that has Form/JSON. 
- No matter what I give in the JSON(valid JSON only), the 'Test' button does not get enabled.

How do I go about resolving these problems? Thanks in advance!

**Comment by mjashanks** [comment]
Hey @rhrish - thank you for reporting these, we'll get them fixed asap.

I've logged them both 
- https://github.com/Budibase/budibase/issues/2964 and 
- https://github.com/Budibase/budibase/issues/2965
